x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/SecurityServerTransportInterceptorTests.java

```diff
@@ -575,9 +589,114 @@ public boolean decRef() {
         assertTrue(exceptionSent.get());
     }
 
+    public void testSendWithCrossClusterAccessHeadersWithUnsupportedLicense() throws Exception {
+        assumeTrue("untrusted remote cluster feature flag must be enabled", TcpTransport.isUntrustedRemoteClusterEnabled());
+
+        // Randomly select license operation mode which is less than ENTERPRISE and not TRIAL.
+        final TestUtils.UpdatableLicenseState unsupportedLicenseState = new TestUtils.UpdatableLicenseState();
+        unsupportedLicenseState.update(
+            randomValueOtherThanMany(
+                mode -> mode.compareTo(Security.CROSS_CLUSTER_ACCESS_FEATURE.getMinimumOperationMode()) >= 0
+                    || mode == License.OperationMode.TRIAL,
+                () -> randomFrom(License.OperationMode.values())
+            ),
+            true,
+            null
+        );
```

Comment:
The reasoning behind this is that I wanted to use "real" license state object rather than a mocked object. This felt closer to the real use case, but on a second thought it makes no difference here.